sql/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -321,7 +322,15 @@ public ColumnContext next() {
             if (ref.column().isTopLevel() == false) {
                 return new ColumnContext(tableInfo, ref, null);
             }
-            return new ColumnContext(tableInfo, ref, ref.position());
+
+            // Tables created with CrateDB < 4.0 don't have any positional information stored inside the meta data so
+            // we must fallback to old behaviour which uses a simple increased iterating based var.
+            Integer position = ref.position();
+            if (position == null) {
+                position = ordinal.intValue();
```

Comment:
Maybe make `ordinal` a primitive int directly?  `++` would cause boxing (although low values are cached, so it's a non issue). The memory footprint should also not increase, due to memory alignment I think a short takes up the same space as an int anyway.